resolver/dns/dns_resolver.go

```diff
@@ -121,6 +127,12 @@ func (b *dnsBuilder) Scheme() string {
 	return "dns"
 }
 
+type netResolver interface {
```

Comment:
Creating a fake or mock for this interface might be the key to building up some coverage. We could also subvert a global function pointer for customAuthorityResolver for the duration of tests.

Alternately we can stand up a UDP socket and fake the DNS protocol to verify that the custom authority catches the traffic.

Reply:
I'm using github.com/miekg/dns to implement my dns server, would it be ok to bring in that dependency for a test?

Reply:
Or perhaps something simple like https://github.com/d-podkorytov/one_dns_go/blob/master/one_dns.go

Reply:
It's preferred not to pull in extra dependency.

I agree with @kriskowal suggestion about having a global function pointer for usage inside `customAuthorityResolver`. We don't really need to have a round trip of DNS request/response, we just need to verify the resolver try to dial to the authority specified in the target string.